### PR TITLE
prefix string_array_t with "utilities"

### PR DESCRIPTION
**Pull request by Karsten1987**
connects to ros2/rmw#100

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
needs to be rebased since https://github.com/ros2/c_utilities/pull/15 has been merged

File: include/c_utilities/types/string_array.h [hunk 1]
```diff
@@ -80,7 +80,7 @@ utilities_get_pre_initialized_string_array(size_t size);
  */
 C_UTILITIES_PUBLIC
 utilities_ret_t
```
Comment (wjwwood) [inline include/c_utilities/types/string_array.h]: This function should also have a [`C_UTILITIES_WARN_UNUSED`](https://github.com/ros2/c_utilities/blob/5711a36d29ba6245d4231ac75f8c5f7e1862c3ba/include/c_utilities/macros.h#L23-L27).

**Comment by Karsten1987** [comment]
@wjwwood Can we agree on the `return UTILITIES_RET_OK` as it is in here: https://github.com/ros2/c_utilities/pull/14/commits/fa1a5d9e5f1bbc0761cf70d5040dfac115b3e6e3#diff-c0cb908ca9bf786997651b24b643070cR49

If so, then i'd go ahead and merge these PRs.

**Comment by wjwwood** [comment]
Yup, lgtm.

**Pull request merged**